src/MongoDB.Driver.Core/Core/Compression/Snappy/Snappy64NativeMethods.cs

```diff
@@ -0,0 +1,96 @@
+    {
+        private static readonly Lazy<LibraryLoader> __libraryLoader;
+        private static readonly Lazy<Delegates64.snappy_compress> __snappy_compress;
+        private static readonly Lazy<Delegates64.snappy_max_compressed_length> __snappy_max_compressed_length;
+        private static readonly Lazy<Delegates64.snappy_uncompress> __snappy_uncompress;
+        private static readonly Lazy<Delegates64.snappy_uncompressed_length> __snappy_uncompressed_length;
+        private static readonly Lazy<Delegates64.snappy_validate_compressed_buffer> __snappy_validate_compressed_buffer;
+
+        static Snappy64NativeMethods()
+        {
+            __libraryLoader = new Lazy<LibraryLoader>(() => new LibraryLoader(GetLibraryLocator), isThreadSafe: true);
+
+            __snappy_compress = CreateLazyForDelegate<Delegates64.snappy_compress>(nameof(snappy_compress));
+            __snappy_max_compressed_length = CreateLazyForDelegate<Delegates64.snappy_max_compressed_length>(nameof(snappy_max_compressed_length));
+            __snappy_uncompress = CreateLazyForDelegate<Delegates64.snappy_uncompress>(nameof(snappy_uncompress));
+            __snappy_uncompressed_length = CreateLazyForDelegate<Delegates64.snappy_uncompressed_length>(nameof(snappy_uncompressed_length));
+            __snappy_validate_compressed_buffer = CreateLazyForDelegate<Delegates64.snappy_validate_compressed_buffer>(nameof(snappy_validate_compressed_buffer));
+        }
+
+        // public static methods
+        public static SnappyStatus snappy_compress(IntPtr input, ulong input_length, IntPtr output, ref ulong output_length)
+        {
+            return __snappy_compress.Value(input, input_length, output, ref output_length);
+        }
+
+        public static ulong snappy_max_compressed_length(ulong input_length)
+        {
+            return __snappy_max_compressed_length.Value(input_length);
+        }
+
+        public static SnappyStatus snappy_uncompress(IntPtr input, ulong input_length, IntPtr output, ref ulong output_length)
+        {
+            return __snappy_uncompress.Value(input, input_length, output, ref output_length);
+        }
+
+        public static SnappyStatus snappy_uncompressed_length(IntPtr input, ulong input_length, out ulong output_length)
+        {
+            return __snappy_uncompressed_length.Value(input, input_length, out output_length);
+        }
+
+        public static SnappyStatus snappy_validate_compressed_buffer(IntPtr input, ulong input_length)
+        {
+            return __snappy_validate_compressed_buffer.Value(input, input_length);
+        }
+
+        // private static methods
+        private static Lazy<TDelegate> CreateLazyForDelegate<TDelegate>(string name)
+        {
+            return new Lazy<TDelegate>(() => __libraryLoader.Value.GetDelegate<TDelegate>(name), isThreadSafe: true);
+        }
+
+         private static string GetLibraryLocator(SupportedPlatform currentPlatform)
```

Comment:
I think thing method should be called `LibraryLocator`.

It doesn't *get* the locator, it *is* the locator.

Reply:
If you want to encapsulate the library locating logic into a class you could use an interface (e.g. `ILibraryLocator`) instead of a delegate.